shared/src/main/scala/org/atnos/eff/Member.scala

```diff
@@ -371,23 +293,23 @@ trait MemberLower6 extends MemberLower7 {
     type Out = Fx3[T, L, R]
 
     def inject[V](m: M[V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      UnionAppendR[Fx1[T], Fx3[L, M, R], V](Union3M(m))
+      Union.appendR(Union.threeM(m))
 
-    def accept[V](union: Union[Fx3[T, L, R], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      union match {
-        case Union3L(t) => UnionAppendL(Union1(t))
-        case Union3M(l) => UnionAppendR(Union3L(l))
-        case Union3R(r) => UnionAppendR(Union3R(r))
+    def accept[V](union: Union[Fx3[T, L, R], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] = union.tagged match {
+      case tagged@UnionTagged(_, index) => (index: @switch) match {
+        case 1 => UnionAppendR(tagged.forget)
+        case _ => UnionAppendL(tagged.xor())
```

Comment:
If `index = 2` (for a `L` effect) then using `xor` makes `2 ^ 2 == 0` and we end-up with a `UnionTagged(_, 0)` for the nested union inside `UnionAppendL`. But all indices for `UnionTagged` should always be `1, 2 or 3`. Am I reading this wrong? I would feel more confident being explicit here or instead make sure we prove why it is correct. This part of the `eff` code worries me a lot because it has a huge potential for runtime errors.